### DISCUSSION TITLE
[Feature Request] Better Button Assignment

### DISCUSSION THREAD

**Discussion by EmDash00** [KeyPuncher/WiinUPro#153] [Ideas]
There's a lot of features of WiinUPro that I feel like are inferior to WiinUSoft and I just wanted to voice some feedback.

It's kind of annoying that in WiinUPro, you have to assign each button separately. It really helped in WiinUSoft that you could visually see each button's assignment via an arrow rather than just have something on text hover. It's really tedious to have to click through each one and assign I really feel like this feature is a downgrade and not an upgrade.

You already are able to show the controller highlighting which button is being pressed, so I think logically you should have a side-by-side screen where the corresponding input on the XBox Controller is shown. For example: I press A on my Wii U Pro Controller which is mapped to B on the emulated XBox Controller. In this case, A would be highlighted on the Wii U Pro Controller, and B would be highlighted on the emulated XBox Controller visualization. On this screen, you should be able to assign buttons as well, integrating visualization of the controller layout and assignment. 

Quick assignment should stay; however, I think it needs to have options for the buttons and sticks. The sticks ought to be quick assignable to other emulated XBox Controller sticks (all four at a time since that would be logical). Quick assignment for buttons should work similarly for groups of buttons. For example, each of the A, B, X, Y buttons should be quick assignable to A, B, X, Y on the emulated XBox Controller.

Finally, there should be default profiles that the controllers come loaded with. The application should just "work" out of the box. It was frustrating to me that I had to go through all this configuration unlike in WiinUSoft where it just worked. The default profiles should cover most use cases so that the user does not need to spend their time playing around with configurations when they just want the controller to work. WiinUSoft had these (at least for the Pro Controller), which could then be adjusted. These should be brought back as they greatly increased usability.

**Comment by KeyPuncher** [comment]
This sort of insight is valuable. However I'd first like to ask, what are you getting out of WiinUPro that you weren't getting out of WiinUSoft? WiinUPro is not supposed to necessarily be the "bigger better" version, instead the two co-exist for different types of users. WiinUSoft is supposed to have that simple straight to the point setup since its targeted at users who want their controller to behave as an Xbox 360 controller (Xbox One would be a nice future upgrade) and that's pretty much all they need.

I think there is still merit to being able to quickly change between what button is being assigned. I'm not sure if that's a hold shift or control then press the button on the controller you want to assign, but that might be more convenient.
I've thought about quick assign for the face buttons just not sure what those would be for non-Xbox controls. Maybe Space, E, R, and Ctrl? Feel like I'd have to research what common keyboard controls are nowadays.

Having a default profile that is bundled into WiinUPro isn't really something I want to put in. Especially for the Xbox controller since users may not even be using that functionality or have the drivers installed. That's what the profile creation feature is for along with setting a default profile to be loaded. There's also default profiles for the Xbox controller packaged with the download so one would just have to point to the desired profile and set it as the default and should be good.
